common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -795,13 +796,25 @@ public void registerExecutor(String appId, ExecutorShuffleInfo executorInfo) {
   }
 
   /**
-   * Close the DB during shutdown
+   * Shutdown mergedShuffleCleaner and close the DB during shutdown
    */
   @Override
   public void close() {
+    if (!mergedShuffleCleaner.isShutdown()) {
+      try {
+        mergedShuffleCleaner.shutdown();
+        boolean terminated = mergedShuffleCleaner.awaitTermination(10L, TimeUnit.SECONDS);
+        if (!terminated) {
+          mergedShuffleCleaner.shutdownNow();
+        }
+      } catch (InterruptedException ignored) {
+        // ignore InterruptedException.
+      }
+    }
```

Comment:
Just wondering should we follow the example provided in Java doc with 2 phase shutdown?
https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ExecutorService.html.

Reply:
```
pool.shutdown(); // Disable new tasks from being submitted
   try {
     // Wait a while for existing tasks to terminate
     if (!pool.awaitTermination(60, TimeUnit.SECONDS)) {
       pool.shutdownNow(); // Cancel currently executing tasks
       // Wait a while for tasks to respond to being cancelled
       if (!pool.awaitTermination(60, TimeUnit.SECONDS))
           System.err.println("Pool did not terminate");
     }
   } catch (InterruptedException ie) {
     // (Re-)Cancel if current thread also interrupted
     pool.shutdownNow();
     // Preserve interrupt status
     Thread.currentThread().interrupt();
   }
```

like above?

Reply:
[7bc9e1c](https://github.com/apache/spark/pull/37624/commits/7bc9e1c6d515b3b7646a6800fcaacc3cd088a1ac) fix this